Source/Libraries/openXDA.APIAuthentication/APIQuery.cs

```diff
@@ -0,0 +1,140 @@
+﻿//******************************************************************************************************
+//  APIQuery.cs - Gbtc
+//
+//  Copyright © 2022, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the Eclipse Public License -v 1.0 (the "License"); you may
+//  not use this file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://www.opensource.org/licenses/eclipse-1.0.php
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/01/2022 - Christoph Lackner
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using GSF.Data;
+
+namespace openXDA.APIAuthentication
+{
+    /// <summary>
+    /// Issues requests to the API using API authentication.
+    /// This class handles the GSF token and authentication
+    /// </summary>
+    public class APIQuery
```

Comment:
Is this class unfinished? I don't see where it's handling any GSF token, and the connection factory is unused.

Reply:
Per our conversation I added the GSF Token logic and removed connectionfactory.
I wasn't sure whether PATCH, DELETE etc need a Token so I am getting on for anything but GET